src/app/shared/services/dynamic-data/adapters/persistedMemory.ts

```diff
@@ -149,11 +164,11 @@ export class PersistedMemoryAdapter {
         delete this.state[flow_type][flow_name][id];
       }
     }
-    // delete all
+    // delete all - mark as empty so it still persists to disk
     else {
-      delete this.state[flow_type][flow_name];
+      this.state[flow_type][flow_name] = {};
```

Comment:
Is it potentially messy to leave empty flows in the DB vs deleting them entirely? Doesn't seem like a major issue either way

Reply:
This is just the in-memory state, so when persisting it iterates over row entries within the hashmap (e.g. `this.state['data_list']['example_flow']['row_id_1']`, so it being empty will also prompt any existing rows to be deleted when persisting.

The previous issue with deleting entirely is that when it compared rows there was no flow to compare against (everything was happening one level less deep than required), and I found adding the empty placeholder worked to fix